### PR TITLE
Add Bloop65

### PR DESCRIPTION
**Pull request by Kwub**
## Description
Add Bloop65 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bloop/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/bloop/config.h:47]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/bloop/config.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/bloop/config.h:61]:
```suggestion
```

File: keyboards/bloop/keymaps/crispyy/keymap.c [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/bloop/keymaps/crispyy/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/bloop/keymaps/kevin_hhkb/keymap.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/bloop/keymaps/kevin_hhkb/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/bloop/rules.mk [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/bloop/rules.mk:41]: Is this using a pro micro? 

If so, then: 
```suggestion
BOOTLOADER = caterina
```
else
```suggestion
BOOTLOADER = atmel-dfu
```

File: keyboards/bloop/rules.mk [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
```
Comment (drashna) [inline keyboards/bloop/rules.mk:56]:
```suggestion
BOOTMAGIC_ENABLE = lite # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no     # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
NKRO_ENABLE = yes       # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
```

File: keyboards/kwub/bloop/config.h [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x0432 // Kwub
```
Comment (drashna) [inline keyboards/kwub/bloop/config.h:22]: This vendor ID is in use and needs to be changed.

https://www.the-sz.com/products/usbid/index.php?v=0x0432&p=&n=

Generally, anything above 0x4200 is not used.

File: keyboards/kwub/bloop/config.h [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/kwub/bloop/config.h:61]:
```suggestion
```

File: keyboards/kwub/bloop/keymaps/crispyy/keymap.c [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kwub/bloop/keymaps/crispyy/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kwub/bloop/keymaps/kevin_hhkb/keymap.c [hunk 10]
```diff
@@ -0,0 +1,23 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/kwub/bloop/keymaps/kevin_hhkb/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/kwub/bloop/rules.mk [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/kwub/bloop/rules.mk:41]:
```suggestion
BOOTLOADER = atmel-dfu
```

File: keyboards/kwub/bloop/rules.mk [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+EXTRAKEY_ENABLE ?= yes		# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no		# Console for debug(+400)
+COMMAND_ENABLE ?= no    	# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE ?= yes			# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  	# Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
```
Comment (drashna) [inline keyboards/kwub/bloop/rules.mk:58]:
```suggestion
BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes   # Mouse keys
EXTRAKEY_ENABLE = yes   # Audio control and System control
CONSOLE_ENABLE = no     # Console for debug
COMMAND_ENABLE = no     # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes       # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no  	# Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = no
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bloop/rules.mk (hunk 5 above)
Comment (Kwub) [reply keyboards/bloop/rules.mk:41]: it is not, i will commit the second, thanks

File: keyboards/bloop/keymaps/crispyy/keymap.c (hunk 3 above)
Comment (Kwub) [reply keyboards/bloop/keymaps/crispyy/keymap.c:1]: added

File: keyboards/bloop/keymaps/kevin_hhkb/keymap.c (hunk 4 above)
Comment (Kwub) [reply keyboards/bloop/keymaps/kevin_hhkb/keymap.c:1]: added

**Comment by Kwub** [comment]
GLP2 Licenses should be resolved with the latest commit, thanks for the note.

**Review by drashna — COMMENTED** [review]
Looks like a couple didn't get added.  May need to push?

**Comment by Kwub** [comment]
Hmm, everything looks up to date on my end

**Comment by Kwub** [comment]
> Looks like a couple didn't get added. May need to push?
Which files are unchanged? Let me check again.

**Comment by drashna** [comment]
The crisppy and kevin_hhkb keymap.c files, specifically.

**Comment by Kwub** [comment]
On my end, all files are up to date and my local is up to date with the remote bloop branch. I think the changes were made in the "added GPL2 licenses" commit (5c37d65). It looks like that commit didn't get checked. How can I resolve this?

**Comment by drashna** [comment]
I'm looking at the files changed on github, as  well as `gh pr checkout` locally.  Both don't show those changes, nor the folder being in the `kwub` folder.  WTF. 

Could you try a `git push --force-with-lease` and see if that fixes things?

**Comment by Kwub** [comment]
Just tried it, it said that everything is up to date. Very weird.

**Comment by drashna** [comment]
Nope, that doesn't seem to have done it.  I ... have no idea.

Aside from maybe rebasing and see if that fixes it:
https://docs.qmk.fm/#/newbs_git_resolving_merge_conflicts?id=rebasing-your-changes

Or go the xkcd route.

**Comment by Kwub** [comment]
Alright, I tried rebasing. Hopefully this works.

**Comment by Kwub** [comment]
I'm not sure what the path forward is here, since the issue still seems unresolved. Should we blow this PR up and try again?

**Comment by drashna** [comment]
Let me see if I can straighten this out

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kwub/bloop/readme.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# Bloop65
+![Bloop65](https://imgur.com/WH56BIu.png)
```
Comment (fauxpark) [inline keyboards/kwub/bloop/readme.md:2]:
```suggestion
![Bloop65](https://imgur.com/WH56BIuh.png)
```

File: keyboards/kwub/bloop/rules.mk [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no     # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes       # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  	# Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
```
Comment (fauxpark) [inline keyboards/kwub/bloop/rules.mk:21]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by Kwub** [comment]
> Let me see if I can straighten this out.
> 
> Edit: There, that's much better.
> 
> for reference, I ran `git rebase upstream/master`, and addressed the merge conflicts as they arose. That can be messy, though.

Ahhh thank you, I really appreciate it.

**Comment by drashna** [comment]
You're very welcome!  And yeah, git can be a nightmare, especially when stuff goes wrong! 

Just make sure that everything looks right.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Looks like you've added the same board twice.

**Comment by Kwub** [comment]
> Looks like you've added the same board twice.

My bad, the folder was still on my local device when I merged. Should be removed now.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**